python/xir/program.py

```diff
@@ -563,6 +562,65 @@ def serialize(self, minimize: bool = False) -> str:
             return strip(res_script)
         return res_script
 
+    @staticmethod
+    def merge(*programs) -> "XIRProgram":
+        """Merges one or more XIR programs into a new XIR program.
+
+        The merged XIR program is formed by concatenating the given XIR programs
+        in the order they are passed to the function. Warnings may be issued for
+        duplicate declarations, gates, includes, operators, and options.
+
+        Args:
+            programs (XIRProgram): XIR programs to merge
+
+        Returns:
+            XIRProgram: the merged XIR program
+
+        Raises:
+            ValueError: if no XIR programs are provided or if at least two XIR
+                programs have different versions or float settings
+        """
+        if len(programs) == 0:
+            raise ValueError("Merging requires at least one XIR program.")
+
+        version = programs[0].version
+        if any(program.version != version for program in programs):
+            raise ValueError("XIR programs with different versions cannot be merged.")
+
+        use_floats = any(program.use_floats for program in programs)
```

Comment:
This means that if any program has `use_floats=True` then the new program will have `use_floats=True`. I think this is fine, but I think you made a case for the opposite being better (having precision kept). 🤔

Reply:
> This means that if any program has `use_floats=True` then the new program will have `use_floats=True`.

Yes, exactly.

> I think this is fine, but I think you made a case for the opposite being better (having precision kept). 🤔

Oh, I thought I argued for having reduced precision!  It seems like the more conservative approach.  Either way, this is trivial to change so I would suggest we submit the PR and update this behaviour later when the requirements are more clear.

Reply:
Haha, I was probably just a bit confused. I though _I_ argued for having reduced precision. 😆